flink-connector-mysql-cdc/src/test/java/com/alibaba/ververica/cdc/connectors/mysql/MySQLSourceTest.java

```diff
@@ -401,6 +414,89 @@ public void go() throws Exception {
             source4.close();
             runThread4.sync();
         }
+
+        {
+            // ---------------------------------------------------------------------------
+            // Step-9: insert partial and alter table
+            // ---------------------------------------------------------------------------
+            final DebeziumSourceFunction<SourceRecord> source5 = createMySqlBinlogSource();
+            final TestSourceContext<SourceRecord> sourceContext5 = new TestSourceContext<>();
+            setupSource(source5, true, offsetState, historyState, true, 0, 1);
+
+            // restart the source
+            final CheckedThread runThread5 =
+                    new CheckedThread() {
+                        @Override
+                        public void go() throws Exception {
+                            source5.run(sourceContext5);
+                        }
+                    };
+            runThread5.start();
+
+            try (Connection connection = database.getJdbcConnection();
+                    Statement statement = connection.createStatement()) {
+
+                statement.execute(
+                        "INSERT INTO products(id, description, weight) VALUES (default, 'Go go go', 111.1)");
+                statement.execute(
+                        "ALTER TABLE products ADD comment_col VARCHAR(100) DEFAULT 'cdc'");
```

Comment:
Could you also add the partial insert and alter column (before checkpoint, and after restore) in IT case? I think we didn't cover record deserialization with the new `DatabaseHistory`.

Reply:
I didn't add this for IT case for now, this PR’s change won't affect the IT case.
And in my  [mysql-cdc-2.0 branch](https://github.com/leonardBang/flink-cdc-connectors/blob/mysql_parallel_nolock/flink-connector-mysql-cdc/src/test/java/com/alibaba/ververica/cdc/connectors/mysql/debezium/reader/BinlogSplitReaderTest.java#L267), I've reconstructed  the  ITCase.